### PR TITLE
Feature/android nav graph

### PR DESCRIPTION
**Pull request by kirstybutler**
**Still work in progress but this PR updates the following:**

- BottomNavigationView with tabs for a map view, list view and settings:
Screenshot 2020-11-19 at 15 05 43
- Creating fragments to replace previous activities to be switched between using the tabs. 
- Using these fragments in the new nav graph:
Screenshot 2020-11-19 at 15 09 16

### CODE REVIEW

File: androidHackathonApp/build.gradle.kts [hunk 1]
```diff
@@ -33,6 +35,15 @@ dependencies {
...
     implementation("androidx.swiperefreshlayout:swiperefreshlayout:1.1.0")
     implementation("androidx.cardview:cardview:1.0.0")
+    implementation("androidx.navigation:navigation-fragment-ktx:2.3.1")
+    implementation("androidx.navigation:navigation-ui-ktx:2.3.1")
+    implementation("org.jetbrains.kotlin:kotlin-stdlib:${rootProject.extra["kotlin_version"]}")
+    implementation("com.google.android.gms:play-services-maps:17.0.0")
+    implementation("androidx.legacy:legacy-support-v4:1.0.0")
+    implementation("androidx.lifecycle:lifecycle-livedata-ktx:2.2.0")
+    implementation("androidx.lifecycle:lifecycle-viewmodel-ktx:2.2.0")
+    implementation ("androidx.navigation:navigation-fragment-ktx:$nav_version")
```
Comment (kirstybutler) [inline androidHackathonApp/build.gradle.kts:43]: oops repeated implementations

File: androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/MainActivity.kt [hunk 2]
```diff
@@ -1,30 +1,21 @@
 package com.kcc.kmmhackathon.androidHackathonApp
```
Comment (KaneCheshire) [inline androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/MainActivity.kt:1]: Nice clean file now!

File: androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/view/SettingsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import com.kcc.kmmhackathon.androidHackathonApp.R
+import com.kcc.kmmhackathon.androidHackathonApp.viewmodel.SettingsViewModel
+
+class SettingsFragment : Fragment() {
+
+    companion object {
+        fun newInstance() = SettingsFragment()
+    }
+
+    private lateinit var viewModel: SettingsViewModel
```
Comment (KaneCheshire) [inline androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/view/SettingsFragment.kt:18]: I think you can do just `private val viewModel: SettingsViewModel by viewModels()`, although you might need another dependency for that

File: lib-find-meal/src/main/java/com/kinandcarta/lib/find/meal/view/FindMealFragment.kt [hunk 4]
```diff
@@ -25,29 +30,39 @@ class FindMealActivity : AppCompatActivity() {
...
 
+        swipeRefreshLayout = view.findViewById(R.id.swipeContainer)
         swipeRefreshLayout.setOnRefreshListener {
             swipeRefreshLayout.isRefreshing = false
             displayMeals(true)
         }
+
         displayMeals(false)
+
+        return view
```
Comment (kirstybutler) [inline lib-find-meal/src/main/java/com/kinandcarta/lib/find/meal/view/FindMealFragment.kt:59]: idk if there should be a space above return or not

File: lib-find-meal/src/main/res/layout/find_meal_fragment.xml [hunk 5]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    tools:context=".view.FindMealFragment">
+
+    <androidx.swiperefreshlayout.widget.SwipeRefreshLayout
```
Comment (KaneCheshire) [inline lib-find-meal/src/main/res/layout/find_meal_fragment.xml:9]: Nice! I've never used one of these so TIL

File: androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/view/SettingsFragment.kt (hunk 3 above)
Comment (kirstybutler) [reply androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/view/SettingsFragment.kt:18]: Yep! This was the next task when moving over to the viewmodels :) although i've never done it this way, but read about it the other day

**Review by dsmith42 — APPROVED** [review]
Looks good 👍

File: lib-find-meal/src/main/res/layout/find_meal_fragment.xml (hunk 5 above)
Comment (kirstybutler) [reply lib-find-meal/src/main/res/layout/find_meal_fragment.xml:9]: The SwipeRefreshLayout? @dsmith42 and @roger-tan did that :)

File: androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/view/SettingsFragment.kt (hunk 3 above)
Comment (kirstybutler) [reply androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/view/SettingsFragment.kt:18]: This is what got auto-generated when creating a` Fragment with ViewModel `using the NavGraph editor

File: lib-find-meal/src/main/res/layout/find_meal_fragment.xml (hunk 5 above)
Comment (dsmith42) [reply lib-find-meal/src/main/res/layout/find_meal_fragment.xml:9]: We yanked it from the kmm tutorial 😃

File: androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/view/SettingsFragment.kt (hunk 3 above)
Comment (KaneCheshire) [reply androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/view/SettingsFragment.kt:18]: Ah yeah some of the autogenerated stuff is a bit old sadly

File: androidHackathonApp/build.gradle.kts [hunk 6]
```diff
@@ -14,6 +15,7 @@ repositories {
 }
 
 val recyclerVersion = "1.1.0"
+val nav_version = "2.3.1"
```
Comment (Gmiller290488) [inline androidHackathonApp/build.gradle.kts:18]: Feel free to ignore - just curious, is the snake case preferred on Android compared to camel case? Would be nice to have them the same way

Comment (kirstybutler) [reply androidHackathonApp/build.gradle.kts:18]: Sure, good point :)

Comment (kirstybutler) [reply androidHackathonApp/build.gradle.kts:18]: I'm not sure which way is preferred for versions, will stick with camel case

**Review by Gmiller290488 — APPROVED** [review]

**Pull request merged**